From a380ec952f138f644d227637eeba90fd1e17f72f Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@openssl.org>
Date: Fri, 19 Apr 2024 10:17:54 -0400
Subject: [PATCH 075/228] Fix missing NULL check in prov_config_test

coverity-1596500 caught a missing null check.  We should never hit it as
the test harness always sets the environment variable, but lets add the
check for safety

Reviewed-by: Tomas Mraz <tomas@openssl.org>
Reviewed-by: Tom Cosgrove <tom.cosgrove@arm.com>
Reviewed-by: Matt Caswell <matt@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/24198)
---
 test/prov_config_test.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/test/prov_config_test.c b/test/prov_config_test.c
index fee2dffdb2..4f0cbc247b 100644
--- a/test/prov_config_test.c
+++ b/test/prov_config_test.c
@@ -83,6 +83,9 @@ static int test_path_config(void)
     char *full_path = NULL;
     int rc;
 
+    if (!TEST_ptr(module_path))
+        return 0;
+
     full_path = OPENSSL_zalloc(strlen(module_path) + strlen(P_TEST_PATH) + 1);
     if (!TEST_ptr(full_path))
         return 0;
-- 
2.39.5 (Apple Git-154)

